src/goto-inspect/goto_inspect_parse_options.cpp

```diff
@@ -0,0 +1,97 @@
+// Author: Fotis Koutoulakis for Diffblue Ltd.
+
+#include "goto_inspect_parse_options.h"
+
+#include <util/config.h>
+#include <util/exception_utils.h>
+#include <util/exit_codes.h>
+#include <util/version.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/read_goto_binary.h>
+#include <goto-programs/show_goto_functions.h>
+
+#include <iostream>
+
+int goto_inspect_parse_optionst::doit()
+{
+  if(cmdline.isset("version"))
+  {
+    std::cout << CBMC_VERSION << '\n';
+    return CPROVER_EXIT_SUCCESS;
+  }
+
+  // Before we do anything else, positional arguments needs to be > 1.
+  // (i.e. a filename has been given).
+  if(cmdline.args.size() < 1)
+  {
+    help();
+    throw invalid_command_line_argument_exceptiont{
+      "failed to supply a goto-binary name",
+      "<input goto-binary> <inspection-option>"};
+  }
```

Comment:
This looks like it should be `> 2` and `< 2` since we require a binary name and another option?

Maybe we also need a code path before this to response to the `--help` option.

Reply:
> This looks like it should be > 2 and < 2 since we require a binary name and another option?

Yes, changed to `> 2`. With the current implementation, if just a binary is given, the behaviour is that the tool doesn't appear to do anything. It's reasonable in my opinion, but not very helpful, as it doesn't indicate to the user whether it succeeded or failed, or what.

> Maybe we also need a code path before this to response to the --help option.

We don't need to do that, as it's handled already by the base class `parse_options_baset::main()`, where it's checking for `h` or `help` before calling `doit()`.